src/pages/CommandIndex/CommandIndexTable/data.tsx

```diff
@@ -0,0 +1,216 @@
+ * @returns A function that takes an argument and returns true only if every
+ * filter in filters returns true for that argument
+ */
+const filtersToFilter = <T,>(
+  filters: ((arg: T) => boolean)[],
+): ((arg: T) => boolean) => {
+  return (arg) => {
+    return filters.every((fn) => fn(arg))
+  }
+}
+
+/**
+ *
+ * @param system - A System object
+ * @returns - The commands associated with the system, each augmented with the
+ * namespace, system name and system version of its parent
+ */
+const commandsAugmenter = (system: System): Command[] => {
+  return system.commands.map(
+    (cmd: Command): Command => ({
+      ...cmd,
+      namespace: system.namespace,
+      systemName: system.name,
+      systemVersion: system.version,
+    }),
+  )
+}
+
+const commandsFromSystems = (
+  systems: System[],
+  includeHidden = false,
+  namespace?: string,
+  systemName?: string,
+  version?: string,
+) => {
+  const hasCommands = (x: System) => !!x.commands && x.commands.length > 0
+  const systemMatcher = [(x: System) => true]
+  const filterHidden = (x: Command) => !x.hidden || includeHidden
+
+  if (version) {
+    systemMatcher.push((x: System) => x.version === version)
+  }
+  if (systemName) {
+    systemMatcher.push((x: System) => x.name === systemName)
+  }
+  if (namespace) {
+    systemMatcher.push((x: System) => x.namespace === namespace)
+  }
+
+  let commands = systems
+    .filter(filtersToFilter(systemMatcher))
+    .filter(hasCommands)
+    .map(commandsAugmenter)
+    .flat()
+    .filter(filterHidden)
+
+  if (includeHidden) {
+    /* make the hidden appear at the top of the list */
+    commands = commands
+      .slice()
+      .sort((a: Command, b: Command) =>
+        a.hidden && !b.hidden ? -1 : !a.hidden && b.hidden ? 1 : 0,
+      )
+  }
+
+  return commands.map(commandMapper)
+}
+
+const useCommands = () => {
+  const [commands, setCommands] = useState<CommandIndexTableData[]>([])
+  const [includeHidden, setIncludeHidden] = useState(false)
+  const { namespace, system_name: systemName, version } = useParams()
+  const { authIsEnabled } = useIsAuthEnabled()
+  const [{ data, error }] = useAxios({
+    url: '/api/v1/systems',
+    method: 'get',
+    withCredentials: authIsEnabled,
+  })
+
+  useEffect(() => {
+    if (data && !error) {
+      setCommands(
+        commandsFromSystems(
+          data,
+          includeHidden,
+          namespace,
+          systemName,
+          version,
+        ),
+      )
+    }
+  }, [data, error, namespace, version, systemName, includeHidden])
+
+  const hiddenOnChange = useCallback(
+    (event: ReactChangeEvent<HTMLInputElement>) => {
+      setIncludeHidden(event.target.checked)
+    },
+    [],
+  )
+
+  return {
+    commands,
+    namespace,
+    systemName,
+    version,
+    includeHidden,
+    hiddenOnChange,
+  }
+}
+
+const useCommandIndexTableColums = () => {
+  return useMemo<Column<CommandIndexTableData>[]>(
+    () => [
+      {
+        Header: 'Command',
```

Comment:
Minor thing, but I'm not sure these headers add much.  On the System table I can kind of take it or leave it, but here page in particular it's a little odd to have a top level header called "Command", and then a column called "Command" that's actually under the "Details" header.  I'd vote to just drop the headers, but if you were going for something in particular and want to tweak it, feel free to give it another shot.

Reply:
Removed. I wasn't in love with it either. ;-)